src/mtype.d

```diff
@@ -10244,15 +10264,95 @@ extern (C++) final class Parameter : RootObject
      *  true = `this` can be used in place of `p`
      *  false = nope
      */
-    final bool isCovariant(const Parameter p) const pure nothrow @nogc @safe
+    final bool isCovariant(bool returnByRef, const Parameter p) const pure nothrow @nogc @safe
     {
         enum stc = STCref | STCin | STCout | STClazy;
-        return !((this.storageClass & stc) != (p.storageClass & stc) ||
+        if ((this.storageClass & stc) != (p.storageClass & stc))
+            return false;
+        return isCovariantScope(returnByRef, this.storageClass, p.storageClass);
+    }
+
+    static bool isCovariantScope(bool returnByRef, StorageClass from, StorageClass to) pure nothrow @nogc @safe
+    {
+        if (from == to)
+            return true;
+
+        /* Shrinking the representation is necessary because StorageClass is so wide
+         * Params:
+         *   returnByRef = true if the function returns by ref
+         *   stc = storage class of parameter
+         */
+        static uint buildSR(bool returnByRef, StorageClass stc) pure nothrow @nogc @safe
```

Comment:
It seems the coverage data is messed up - the enums further on are not executable code.